astropy/table/sorted_array.py

```diff
@@ -0,0 +1,286 @@
+import numpy as np
+
+class SortedArray(object):
+    '''
+    Implements a sorted array container using
+    a list of numpy arrays.
+
+    Parameters
+    ----------
+    lines : `Table`
```

Comment:
Yeah, splitting it that way is probably best. It's necessary to have the row column bundled with the value columns in a Table before performing `lexsort`, but after that we can split it up again into `data` and `row_index` when passing to the engine `__init__`.

Currently `data` is a list of ndarrays rather than Column objects. I could definitely turn it into a Table if that makes the code easier to read/maintain, although my initial thought was that an ndarray will be faster, e.g. since Column has a custom `__setitem__`.